packages/caliper-core/test/worker/caliper-worker.js

```diff
@@ -161,5 +190,101 @@ describe('Caliper worker', () => {
+                mockStats.getCumulativeTxStatistics.returns({});
+
+                worker.internalTxObserver.getCurrentStatistics = () => mockStats;
+
+                mockWorkload.submitTransaction.callsFake(async () => {
+                    submittedTx += 1;
+                    finishedTx += 1;
+                    return Promise.resolve();
+                });
+
+                await worker.executeRound(mockTestMessage);
+
+                sinon.assert.callCount(mockWorkload.submitTransaction, numberOfTxs);
+                sinon.assert.calledOnce(deactivateMethod);
+                sinon.assert.calledOnce(mockRate.end);
+                sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+                sinon.assert.called(mockConnector.releaseContext);
+            });
+        });
+
+        it('should execute the round for a specified duration', async function() {
+            this.timeout(5000); // Increase the timeout for this test
+            const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+            await worker.prepareTest(mockTestMessage);
+            mockWorkload.submitTransaction.resolves();
+            mockTestMessage.getRoundDuration.returns(1); // duration in seconds
+
+            await worker.executeRound(mockTestMessage);
+
+            sinon.assert.calledOnce(deactivateMethod);
+            sinon.assert.calledOnce(mockRate.end);
+            sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+            sinon.assert.called(mockConnector.releaseContext);
+        });
+
+
+        it('should handle errors during the prepareTest phase', async () => {
+            const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+            const errorMessage = 'Initialization error';
+            mockConnector.getContext.rejects(new Error(errorMessage));
+            mockTestMessage.getRoundIndex.returns(1);
+            mockTestMessage.getWorkloadSpec.returns({ module: 'test/workload' });
+            mockTestMessage.getWorkerArguments.returns([]);
+
+            await worker.prepareTest(mockTestMessage).should.be.rejectedWith(errorMessage);
+
+            sinon.assert.calledOnce(mockConnector.getContext);
+            sinon.assert.calledOnce(logwarningMethod);
+        });
+
+        it('should not submit transactions after the duration ends', async function() {
+            this.timeout(5000);
+
+            const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+            await worker.prepareTest(mockTestMessage);
+
+            const clock = sinon.useFakeTimers();
+            mockWorkload.submitTransaction.resolves();
+
+            mockTestMessage.getRoundDuration.returns(1);
+            mockTestMessage.getNumberOfTxs.returns(null);
+
+            const executePromise = worker.executeRound(mockTestMessage);
+
+            await clock.tickAsync(1000); // Advance time by 1 second
+            await Promise.resolve();
```

Comment:
not sure what this line is doing here could you enlighten me please ?

Reply:
After advancing the virtual clock forward with `clock.tickAsync`, which triggers only the macro-tasks, I needed to ensure that all microtasks also get processed. I did this by using `await Promise.resolve();` to yield control back to the event loop.

Reply:
Be good to add a comment here explaining that.